R/chat.R

```diff
@@ -386,19 +386,32 @@ Chat <- R6::R6Class(
     #'   that yields strings. While iterating, the generator will block while
     #'   waiting for more content from the chatbot.
     #' @param ... The input to send to the chatbot. Can be strings or images.
-    stream = function(...) {
+    #' @param content Whether the stream should yield only `"text"` or `"all"`
+    #'   content types created during the round. When `content = "all"`,
+    #'   `stream()` yields [Content] objects.
+    stream = function(..., content = c("text", "all")) {
+      content <- arg_match(content)
       turn <- user_turn(...)
-      private$chat_impl(turn, stream = TRUE, echo = "none")
+      private$chat_impl(turn, content = content, stream = TRUE, echo = "none")
```

Comment:
Coming back to this, I think `content` makes sense as the user-facing option in `$stream()` and `$stream_async()`, but inside `$chat_impl*() and `$submit_turns*()` I'd much rather call this `yield_content`. 

I find `yield_content` a lot clearer and more approachable in those contexts and reinforces that there are two different places where results may be surfaced and they have different options.

Reply:
I made this change in 803b94197819f73aba9c8f1c8d7d5fba9d4d4b45, but can revert if desired. The primary motivation is that `content` is a confusing variable name when we're working with the API response -- it's not the content of the response, nor a `Content` instance, but rather a third thing which is actually just an option that only applies to what results are yielded.